packages/hub/src/components/NavigationBlocker/NavigationBlockerProvider.tsx

```diff
@@ -0,0 +1,25 @@
+import { FC, PropsWithChildren, useReducer } from "react";
+
+import { NavigationBlockerContext, navigationBlockerReducer } from "./context";
+import { InterceptedLinkModal } from "./InterceptedLinkModal";
+
+export const NavigationBlockerProvider: FC<PropsWithChildren> = ({
```

Comment:
I think the name "NavigationBlocker" is kind of vague. Maybe:
- UnsavedChangesGuard
- DirtyStateNavigation Intercepter/Guard

Reply:
I talked to Claude and it suggested better names for this and for hooks.